code/game/objects/items/tools/weldingtool.dm

```diff
@@ -0,0 +1,345 @@
+				user.transferItemToLoc(src, F, TRUE)
+			F.weldtool = src
+			add_fingerprint(user)
+			to_chat(user, "<span class='notice'>You add a rod to a welder, starting to build a flamethrower.</span>")
+			user.put_in_hands(F)
+		else
+			to_chat(user, "<span class='warning'>You need one rod to start building a flamethrower!</span>")
+
+/obj/item/weldingtool/ignition_effect(atom/A, mob/user)
+	if(welding && remove_fuel(1, user))
+		. = "<span class='notice'>[user] casually lights [A] with [src], what a badass.</span>"
+	else
+		. = ""
+
+/obj/item/weldingtool/largetank
+	name = "industrial welding tool"
+	desc = "A slightly larger welder with a larger tank."
+	icon_state = "indwelder"
+	max_fuel = 40
+	materials = list(MAT_GLASS=60)
+	origin_tech = "engineering=2;plasmatech=2"
+
+/obj/item/weldingtool/largetank/cyborg
+	name = "integrated welding tool"
+	desc = "An advanced welder designed to be used in robotic systems."
+	toolspeed = 0.5
+
+/obj/item/weldingtool/largetank/flamethrower_screwdriver()
+	return
+
+
+/obj/item/weldingtool/mini
+	name = "emergency welding tool"
+	desc = "A miniature welder used during emergencies."
+	icon_state = "miniwelder"
+	max_fuel = 10
+	w_class = WEIGHT_CLASS_TINY
+	materials = list(MAT_METAL=30, MAT_GLASS=10)
+	change_icons = 0
+
+/obj/item/weldingtool/mini/flamethrower_screwdriver()
+	return
+
+/obj/item/weldingtool/abductor
+	name = "alien welding tool"
+	desc = "An alien welding tool. Whatever fuel it uses, it never runs out."
+	icon = 'icons/obj/abductor.dmi'
+	icon_state = "welder"
+	toolspeed = 0.1
+	light_intensity = 0
+	change_icons = 0
+	origin_tech = "plasmatech=5;engineering=5;abductor=3"
+
+/obj/item/weldingtool/abductor/process()
+	if(get_fuel() <= max_fuel)
+		reagents.add_reagent("welding_fuel", 1)
+	..()
+
+/obj/item/weldingtool/hugetank
+	name = "upgraded industrial welding tool"
+	desc = "An upgraded welder based of the industrial welder."
+	icon_state = "upindwelder"
+	item_state = "upindwelder"
+	max_fuel = 80
+	materials = list(MAT_METAL=70, MAT_GLASS=120)
+	origin_tech = "engineering=3;plasmatech=2"
+
+/obj/item/weldingtool/experimental
+	name = "experimental welding tool"
+	desc = "An experimental welder capable of self-fuel generation and less harmful to the eyes."
+	icon_state = "exwelder"
+	item_state = "exwelder"
+	max_fuel = 40
+	materials = list(MAT_METAL=70, MAT_GLASS=120)
+	origin_tech = "materials=4;engineering=4;bluespace=3;plasmatech=4"
+	var/last_gen = 0
+	change_icons = 0
+	can_off_process = 1
+	light_intensity = 1
+	toolspeed = 0.5
+	var/nextrefueltick = 0
+
+/obj/item/weldingtool/experimental/brass
+	name = "brass welding tool"
+	desc = "A brass welder that seems to constantly refuel itself. It is faintly warm to the touch."
+	resistance_flags = FIRE_PROOF | ACID_PROOF
+	icon_state = "brasswelder"
+	item_state = "brasswelder"
+
+
+/obj/item/weldingtool/experimental/process()
+	..()
+	if(get_fuel() < max_fuel && nextrefueltick < world.time)
+		nextrefueltick = world.time + 10

```

Comment:
Can this break at midnight?

Reply:
I don't know, this is just the way it already is.

Reply:
world.time does not suffer from problems with midnight rollover.